drizzlepac/outputimage.py

```diff
@@ -530,32 +545,40 @@ def writeFITS(self, template, sciarr, whtarr, ctxarr=None,
                 # header to Primary header...
                 if dqhdr:
                     for _card in dqhdr.cards:
-                        if ( (_card.keyword not in RESERVED_KEYS) and
+                        if ((_card.keyword not in RESERVED_KEYS) and
                              _card.keyword not in hdu.header):
                             hdu.header.append(_card)
                 hdu.header['filename'] = self.outcontext
+
                 if self.wcs:
-                    pre_wcs_kw = self.find_kwupdate_location(hdu.header,'CD1_1')
+                    pre_wcs_kw = self.find_kwupdate_location(hdu.header, 'CD1_1')
                     # Update WCS Keywords based on PyDrizzle product's value
                     # since 'drizzle' itself doesn't update that keyword.
-                    addWCSKeywords(self.wcs,hdu.header, blot=self.blot,
+                    addWCSKeywords(self.wcs, hdu.header, blot=self.blot,
                                    single=self.single, after=pre_wcs_kw)
 
+                # explicitly remove EXTNAME, EXTVER from header - they are not allowed
+                for kw in RESERVED_KEYS[-2:]:
+                    if kw in hdu.header:
+                        del hdu.header[kw]
```

Comment:
Same reasoning as for `sciarr` above, this is not necessary: code on lines `#541-543` insures that `hdu` is not going to contain `'EXTNAME'` and `'EXTVER'`. Also, replace line `#543` with:
```python
hdu = fits.PrimaryHDU(data= _ctxarr, header=prihdu.header)
```

Reply:
Included as part of 4b565a7